presto-docs/src/main/sphinx/develop/presto-native.rst

```diff
@@ -0,0 +1,96 @@
+===========================
+Presto Native - Prestissimo
+===========================
+
+Prestissimo is a C++ drop-in replacement for Presto workers based on the Velox
+library. It implements the same RESTful API as Java workers using the Proxygen
+C++ HTTP framework. Since communication with the Java coordinator and across
+workers in only done via the REST APIs, Prestissimo does not use JNI and it is
+commonly deployed in such a way as to avoid having a JVM process on worker
+nodes.
+
+Prestissimo's codebase is located at `presto-native-execution
+<https://github.com/prestodb/presto/tree/master/presto-native-execution>`_.
+
+
+Endpoints
+---------
+
+HTTP endpoints related to tasks are registered to Proxygen in
+`TaskResource.cpp`. Important endpoints implemented include:
+
+* POST: v1/task: This processes a `TaskUpdateRequest`
+* GET: v1/task: This returns a serialized `TaskInfo` (used for comprehensive
+  metrics, may be reported less frequently) 
+* GET: v1/task/status: This returns
+  a serialized `TaskStatus` (used for query progress tracking, must be reported
+  frequently)
+
+Other HTTP endpoints include:
+
+* POST: v1/memory
+  * This just reports memory, but no assignments are adjusted unlike in Java
+    workers.
+* GET: v1/info
+* GET: v1/status
+
+The request/response flow of Prestissimo is identical to Java workers. The
+tasks or new splits are registered via `TaskUpdateRequest`. Resource
+utilization and query progress are sent to the coordinator via task endpoints.
+
+
+Remote Function Execution
+-------------------------
+
+Prestissimo supports remote execution of scalar functions. This feature is
+useful for cases when the function code is not written in C++, or if for
+security or flexibility reasons the function code cannot be linked to the same
+executable as the main engine.
+
+Remote function signatures need to be provided using a JSON file, following
+the format implemented by `JsonFileBasedFunctionNamespaceManager
+<https://github.com/prestodb/presto/blob/master/presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/json/JsonFileBasedFunctionNamespaceManager.java>`_.
+The following properties allow the configuration of remote function execution:
+
+``remote-function-server.signature.files.directory.path``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    * **Type:** ``string``
+    * **Default value:** ``""``
+
+    The local filesystem path where json files containing remote function
```

Comment:
"JSON file" is capitalized in the previous paragraph but not in this paragraph. Consider standardizing how "json file" is capitalized throughout the document for consistency.